### ISSUE TITLE
Provide styling for `input[type="file"]`

### ISSUE THREAD

**Issue by Lily2point0** [Financial-Times/o-forms#199]
It would be great to have a class for file inputs in forms.

The current default Chrome styling is:
![screen shot 2018-03-27 at 12 31 01](https://user-images.githubusercontent.com/3882381/37964879-f03a2b2e-31ba-11e8-884e-52cd2fcc8ac5.png)

and, when the `o-forms__text` class is applied:
![screen shot 2018-03-27 at 12 30 47](https://user-images.githubusercontent.com/3882381/37964903-05e74330-31bb-11e8-917d-679b0bc108d0.png)

The "Choose file" button should reflect `o-button` styling if possible.

**Comment by gvonkoss** [comment]
@Lily2point0 A couple of the designers and I are going to be discussing some of the issues for `o-forms`, is this still relevant?

**Comment by gvonkoss** [comment]
We'll be waiting for UXD feedback to add this feature, for now.

**Comment by notlee** [comment]
[Helen H would also have found this useful (see Slack)](https://financialtimes.slack.com/archives/C02FU5ARJ/p1569347196032500).
![image](https://user-images.githubusercontent.com/10405691/65536683-d13b6900-defb-11e9-816f-8a9d5672943c.png)

**Comment by notlee** [comment]
[Ben F would also have found this useful (see Slack)](https://financialtimes.slack.com/archives/C02FU5ARJ/p1611313670008400).

**Comment by chee** [comment]
i'm adding the accessibility label to remind us to ensure we test this with screenreaders and Dragon

**Comment by chee** [comment]
- [ ] get a design for this

**Comment by chee** [comment]
after reading this https://scottaohara.github.io/a11y_styled_form_controls/src/file-upload/, i think i'd prefer if we wrote some guidance on why we don't provide styles for the file upload, rather than trying to make it look like an o-button

**Comment by JakeChampion** [comment]
> after reading [this](https://scottaohara.github.io/a11y_styled_form_controls/src/file-upload/), i'd prefer if we wrote some guidance on why we don't provide styles for the file upload, rather than trying to make it look like an o-button

We can do that, yes 👍 

I've spoken with Ben F and they were asking about file upload for a personal project and not a work project, which means we could pull this out of our issues to work on during the current cycle if we want to.

**Comment by notlee** [comment]
>We can do that, yes 👍
+1 totally a valid and good option

**Comment by JakeChampion** [comment]
Something like this perhaps?

>Origami does not provide a custom style for file inputs because it may be confusing for users who are using assistive technologies. This is due to the fact that assistive technologies act differently with how they announce file inputs, and having a custom style which doesn't match the announcements being made may confuse users.

That is a rewrite from a note on the [link]([this](https://scottaohara.github.io/a11y_styled_form_controls/src/file-upload/)) Chee posted earlier, which was written by Scott O'Hara.

The original note is below:
>Due to browsers exposing native file uploads in different ways (both visually and with what's announced), a single design for a file upload may be confusing for some users if the visual style doesn't match expected announcements.

**Comment by notlee** [comment]
@JakeChampion sounds good but I think we should get the design team onboard to help ensure individual teams don't build their own anyway. We had discussed potentially adding styles around the file upload to make it FT-y, for example like GDS use these error styles without modifying the input itself:

![Screenshot 2021-03-16 at 12 47 40](https://user-images.githubusercontent.com/10405691/111311237-de9b6480-8655-11eb-83b3-a1bd39616fd6.png)

**Comment by chee** [comment]
@Financial-Times/designers hello! do you have any thoughts on styling for file inputs?

**Comment by JakeChampion** [comment]
Styling the file input button is now possible with a non-prefixed selector. Currently supported in stable Firefox, nightly Chrome and nothing else
Here's a demo showing the file input button being styled like an o-button -- https://jsbin.com/putimajapa/edit?html,css,output
![image](https://user-images.githubusercontent.com/1569131/114163344-fa76eb00-9921-11eb-967a-a6f7a7a33556.png)

<details><summary>Demo HTML</summary>
<pre><code><!DOCTYPE html>
<html>
<head>
  <meta charset="utf-8">
  <meta name="viewport" content="width=device-width">
  <title>JS Bin</title>
  <link rel="stylesheet" href="//www.ft.com/__origami/service/build/v2/bundles/css?modules=o-buttons%406.2.0%3A%2Fdemos%2Fsrc%2Fdemo.scss%2Co-fonts%40%5E4.0.0%2Co-normalise%40%5E2.0.0&amp;brand=master">
  <style>
    /* This is the same styles as: o-buttons o-buttons--primary o-buttons--big */
/* https://developer.mozilla.org/en-US/docs/Web/CSS/::file-selector-button */
input[type=file]::file-selector-button {
  appearance: none;
  background-clip: border-box;
  background-color: rgb(13, 118, 128);
  background-size: 40px 40px;
  border-bottom-color: rgba(0, 0, 0, 0);
  border-bottom-left-radius: 0px;
  border-bottom-right-radius: 0px;
  border-bottom-style: solid;
  border-bottom-width: 1px;
  border-left-color: rgba(0, 0, 0, 0);
  border-left-style: solid;
  border-left-width: 1px;
  border-right-color: rgba(0, 0, 0, 0);
  border-right-style: solid;
  border-right-width: 1px;
  border-top-color: rgba(0, 0, 0, 0);
  border-top-left-radius: 0px;
  border-top-right-radius: 0px;
  border-top-style: solid;
  border-top-width: 1px;
  box-sizing: border-box;
  color: rgb(255, 255, 255);
  cursor: pointer;
  display: inline-block;
  font-family: MetricWeb, sans-serif;
  font-size: 16px;
  font-weight: 600;
  line-height: 16px;
  margin-bottom: 0px;
  margin-left: 0px;
  margin-right: 10px;
  margin-top: 0px;
  min-height: 40px;
  min-width: 80px;
  overflow: visible;
  overflow-x: visible;
  overflow-y: visible;
  padding-bottom: 11px;
  padding-left: 20px;
  padding-right: 20px;
  padding-top: 11px;
  text-align: center;
  text-decoration: rgb(255, 255, 255);
  text-decoration-color: rgb(255, 255, 255);
  text-decoration-line: none;
  text-decoration-style: solid;
  text-decoration-thickness: auto;
  text-rendering: optimizelegibility;
  text-transform: none;
  transition-delay: 0s, 0s, 0s;
  transition-duration: 0.3s, 0.15s, 0.15s;
  transition-property: background-color, color, border-color;
  transition-timing-function: ease, ease-out, ease-out;
  user-select: none;
  vertical-align: middle;
-moz-osx-font-smoothing: grayscale;
}
  </style>
</head>
<body>
  <input type=file />
</body>
</html></code></pre></details>

**Comment by chee** [comment]
@Financial-Times/designers Would you be happy with the o-button style for file inputs that Jake has demo'd here? (it'll fallback to the current unstyled files for browsers that don't support it)

![image](https://user-images.githubusercontent.com/178266/115032177-8eaff780-9eb8-11eb-9fc1-fc0401cdd583.png)

**Comment by emmsielu** [comment]
Hello Origami - sorry about the slowness of the response. I'm not noticing the emails enough (they're in my personal email - I should change it!) and I don't know if anyone else is aware? (Although, I think they're added?) Is there a github integration for slack, so i can get this to push something to our design ops channel when "Designers" gets used here?

Also, just looking now :) The o-button looks fine. The one thing I noticed, and maybe it's me??? - in the JSBin example, when I click the button, and after you select an item in the pop up dialogue, the button edges have a corner radius on them?

**Comment by balfourd** [comment]
This looks great - definitely useful for some forms I'll be working on in Spark. Looks great ❤️

**Comment by JakeChampion** [comment]
I've made a pull-request for this -- https://github.com/Financial-Times/o-forms/pull/394

**Comment by JakeChampion** [comment]
This was added in [version 8.5.0](https://registry.origami.ft.com/components/o-forms@8.5.0/#demo-file-input) 🎉 
We style the native file input, with the ::file-selector-button pseudo-element, which means the interactions for assistive technologies such as screen readers and speech input have not changed 😄 

![Screenshot of the file input demo from the Origami Registry](https://user-images.githubusercontent.com/1569131/115273914-0a6ba780-a138-11eb-81a8-3c96f9d13c9f.png)
